conans/client/build/autotools_environment.py

```diff
@@ -241,38 +241,38 @@ def _configure_link_flags(self):
         """Not the -L"""
         ret = copy.copy(self._deps_cpp_info.sharedlinkflags)
         ret.extend(self._deps_cpp_info.exelinkflags)
-        ret.extend(format_frameworks(self._deps_cpp_info.frameworks, compiler=self._compiler))
-        ret.extend(format_framework_paths(self._deps_cpp_info.framework_paths, compiler=self._compiler))
-        arch_flag = architecture_flag(compiler=self._compiler, os=self._os, arch=self._arch)
+        ret.extend(format_frameworks(self._deps_cpp_info.frameworks, self._conanfile.settings))
+        ret.extend(format_framework_paths(self._deps_cpp_info.framework_paths, self._conanfile.settings))
+        arch_flag = architecture_flag(self._conanfile.settings)
         if arch_flag:
             ret.append(arch_flag)
 
-        sysf = sysroot_flag(self._deps_cpp_info.sysroot, win_bash=self._win_bash,
-                            subsystem=self.subsystem,
-                            compiler=self._compiler)
+        sysf = sysroot_flag(self._deps_cpp_info.sysroot, self._conanfile.settings,
+                            win_bash=self._win_bash,
+                            subsystem=self.subsystem)
         if sysf:
             ret.append(sysf)
 
         if self._include_rpath_flags:
             os_build, _ = get_build_os_arch(self._conanfile)
             if not hasattr(self._conanfile, 'settings_build'):
                 os_build = os_build or self._os
-            ret.extend(rpath_flags(os_build, self._compiler, self._deps_cpp_info.lib_paths))
+                ret.extend(rpath_flags(self._conanfile.settings, self._deps_cpp_info.lib_paths))
```

Comment:
This doesn't make much sense to me. Now the ``os_build`` is unused?

Reply:
what do you mean? now we don't pass selected individual values of settings, we just pass all the settings together. the function `rpath_flags` (and others) queries **os_build** inside.

Reply:
``os_build = os_build or self._os`` is unused now. ``os_build`` is a local variable.

Reply:
this seems to be a new code, I didn't see this one when added the feature, e.g. it appeared only after the merge and solving all the conflicts. I think I can remove that if it's not used somewhere else below (?) and if it's not the global/class variable changed here.

Reply:
Calling @jgsogo to check this, as this code is part of the new contexts.

Reply:
This changed here (after this PR was created) https://github.com/conan-io/conan/pull/6916/files#diff-592f24044b25da4ad2488c3887be2261L253. 

For this concrete call, it would be needed all the `self._conanfile` in the `rpath_flags()` function to be able to reproduce the behavior: we need access to the _host_ settings, but the call to `get_build_os_arch()` is using settings_build too.

So probably, all the calls that started to take the `settings` object now need the `conanfile`... or they can come back to use selected individual values of settings and we can make that refactor afterward.

Reply:
well, for conanfile, it might not be always available (?), but I think I can try them to accept conanfile then and see how is it going

Reply:
I changed `rpath_flags` to accept `os_build` for now. we can re-factor in another PR, if necessary.

Reply:
👍